src/ObjCRuntime/Runtime.cs

```diff
@@ -1684,12 +1686,26 @@ static Type LookupINativeObjectImplementation (IntPtr ptr, Type target_type, Typ
 					// native objects and NSObject instances.
 					throw ErrorHelper.CreateError (8004, $"Cannot create an instance of {implementation.FullName} for the native object 0x{ptr:x} (of type '{Class.class_getName (Class.GetClassForObject (ptr))}'), because another instance already exists for this native object (of type {o.GetType ().FullName}).");
 				}
-				return (T?) ConstructNSObject<T> (ptr, implementation, MissingCtorResolution.ThrowConstructor1NotFound);
+				var rv = (T?) ConstructNSObject<T> (ptr, implementation, MissingCtorResolution.ThrowConstructor1NotFound);
+				if (owns)
+					TryReleaseINativeObject (rv);
+				return rv;
 			}
 
 			return ConstructINativeObject<T> (ptr, owns, implementation, MissingCtorResolution.ThrowConstructor2NotFound);
 		}
 
+		static void TryReleaseINativeObject (INativeObject? obj)
+		{
+			if (obj is null)
```

Comment:
obj is null is redundant, as the `is NativeObject` does a null check already.

Reply:
The `is NativeObject` check throws if it fails. The null check doesn't.